darwin/dataset/download_manager.py

```diff
@@ -464,14 +473,33 @@ def _fetch_multiple_files(path: Path, response: requests.Response) -> None:
         path = dir_path / filename
         response = requests.get(url, stream=True)
         if response.ok:
-            _write_file(path, response)
+            _write_file(path, response, transform_file_function)
         else:
             raise Exception(
                 f"Request to ({url}) failed. Status code: {response.status_code}, content:\n{get_response_content(response)}."
             )
 
 
-def _write_file(path: Path, response: requests.Response) -> None:
+def _write_file(path: Path, response: requests.Response, transform_file_function=None) -> None:
     with open(str(path), "wb") as file:
         for chunk in response:
             file.write(chunk)
+    if transform_file_function is not None:
+        transform_file_function(path)
+
+
+def _rg16_to_grayscale(path):
+    # Custom 16bit grayscale encoded on (RG)B channels
+    # into regular 8bit grayscale
+
+    image = Image.open(path)
+    image_2d_rgb = np.asarray(image)
+
+    image_2d_r = np.uint16(image_2d_rgb[:, :, 0]) << 8
+    image_2d_g = np.uint16(image_2d_rgb[:, :, 1])
+
+    image_2d_gray = np.bitwise_or(image_2d_r, image_2d_g)
+    image_2d_gray = image_2d_gray / (1 << 16) * 255
+
+    new_image = Image.fromarray(np.uint8(image_2d_gray), mode="L")
+    new_image.save(path)
```

Comment:
This is doing a decent job of the RGB -> Grayscale conversion based on focussing on R and G is better than the mean luminescence of R,G, and B, but if you find the outputs are lacking, the coeffiecients below are useful for grayscale transformation:

If so, here they are:
```
grey = 0.2989 * R + 0.5870 * G + 0.1140 * B
```

Reply:
This is not RGB input though. This is our custom format that encodes 16 grayscale image on first 2 bytes of RGB tuple. "B" channel is all zeros in that case. Those files are generated by our backend here: https://github.com/v7labs/busybee/blob/master/dicom-extractor/dicom-extractor.py#L72

Reply:
Ah right, didn't realise this was a DICOM input.   Yeah, in that case my earlier comment is not at all applicable!